src/components/drawer/drawer.js

```diff
@@ -0,0 +1,162 @@
+import { getUniqueId } from '@brightspace-ui/core/helpers/uniqueId.js';
+import { heading3Styles } from '@brightspace-ui/core/components/typography/styles.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { LocalizeLabsElement } from '../../../helpers/localizeLabsElement.js';
+
+const mediaQueryList = window.matchMedia('(max-width: 615px), (max-height: 420px) and (max-width: 900px)');
+
+/**
+ * Modal element that is anchored to one side of the viewport or parent element.
+ * @fires d2l-drawer-before-close - Dispatched with bound action value before the dialog is closed for any reason, providing an opportunity to prevent the closing.
+ * @slot - Default slot for content inside drawer
+ * @slot footer - Slot for footer content such as workflow buttons. Apply attribute `drawer-close` to workflow buttons to have them automatically close the drawer with the action's bound `value`.
+ */
+class Drawer extends LocalizeLabsElement(DrawerMixin(LitElement)) {
+
+	static get properties() {
+		return {
+			/**
+             * Aria label for default provided close button
+             */
+			closeButtonLabel: { type: String, attribute: 'close-button-label' },
+			/**
+             * Disables scroll locking if true
+             */
+			lockScroll: { type: Boolean, attribute: 'lock-scroll', reflect: true },
+			/**
+             * Control drawer size. Can use predefined values (sm, md, lg, full), or set drawer body size with any px or % values
+             * Controls width for left and right positions and height for top and bottom.
+             */
+			size: { type: String },
+			/**
+             * Trap user focus if true
+             */
+			trapFocus: { type: Boolean, reflect: true },
+			/**
+             * Hides close button if true, can still be closed via escape key or clicking out
+             */
+			hideCloseButton: { type: Boolean, attribute: 'hide-close-button' },
+			/**
+             * Hides header if true
+             */
+			hideHeader: { type: Boolean, attribute: 'hide-header' },
+			_hasFooterContent: { type: Boolean, attribute: false },
+		};
+	}
+
+	static get styles() {
+		return [ drawerStyles, drawerHeaderStyles, drawerFooterStyles, heading3Styles ];
+	}
+
+	constructor() {
+		super();
+		this.hideCloseButton = false;
+		this.hideHeader = false;
+		this.lockScroll = true;
+		this.size = 'md';
+		this.trapFocus = true;
+		this._hasFooterContent = false;
+		this._handleResize = this._handleResize.bind(this);
+		this._labelId = undefined;
+		this._textId = undefined;
+	}
+
+	get closeButton() {
+		return !this.hideCloseButton
+			? html`
+                <d2l-button-icon
+                    id="drawer-default-close"
+                    icon="tier1:close-small"
+                    text="${this.localize('components.drawer.close')}"
+                    @click=${this._abort}
+                    tabindex=100>
+                </d2l-button-icon>`
+			: nothing;
+	}
+	get content() {
```

Comment:
Is it intended for `content` and `header` to be public properties? Normally if we want to break up the rendering we will create internal helper methods, `_renderHeader() { ... }`.